### PR TITLE
Adding support for Mermaid diagrams and MathJax Rendering

### PR DESCRIPTION
**Pull request by nanspro**
- Added support for `mermaid` (_gantt diagram, flowchart, sequence diagram, class diagram, git graph_)

- Added support for `mathJax` and `katex` to render math beautifully, similar to what hackmd uses

Fixes https://github.com/status-im/status.im/issues/200

### CODE REVIEW

**Comment by nanspro** [comment]
This PR is linked to https://github.com/status-im/status.im-partials/pull/8

**Comment by nanspro** [comment]
Examples of rendering some diagrams (by changing themes we can generate them in different ways)
screenshot_2019-02-19 status a mobile client for ethereum 2 0

File: _config.dev.yml [hunk 1]
```diff
@@ -37,8 +37,27 @@ github: staus-im
...
+
+math:
+  engine: 'mathjax' # or 'katex'
+  mathjax:
+    src: custom_mathjax_source
+    config:
+      # MathJax config
+  katex:
+    css: custom_css_source
+    js: custom_js_source # not used
```
Comment (richard-ramos) [inline _config.dev.yml]: These values: `custom_matchjax_source`, `custom_css_source` and `custom_js_source` , aren't these values supposed to be changed to real data? According to the documentation, this config will be merged with the plugin default configuration, and in the default config, these values are URLs.

File: _config.dev.yml [hunk 2]
```diff
@@ -37,8 +37,27 @@ github: staus-im
 
 graphviz:
   enable: true
+  
+# mermaid chart
+mermaid: ## mermaid url https://github.com/knsv/mermaid
```
Comment (richard-ramos) [inline _config.dev.yml]: Step 3 from this documentation is missing?
https://github.com/webappdevelp/hexo-filter-mermaid-diagrams#step3-include-mermaidjs-in-pug-or-ejs

**Comment by nanspro** [comment]
screenshot from 2019-02-22 18-35-54
screenshot from 2019-02-22 18-34-14

File: _config.dev.yml (hunk 1 above)
Comment (nanspro) [reply _config.dev.yml]: These values are just there to show that you can add your custom_css and custom_js but as of now  status doesn't need it so i am removing it form there.
@richard-ramos the issue is adding support for math,mermaid,flow,sequence and we are not using any of them right now but whenever we need to use it in feature, just change the values and write the syntax.

**Comment by nanspro** [comment]
@andytudhope @richard-ramos  could any of you review it again?

**Comment by andytudhope** [comment]
This lgtm, wdyt @richard-ramos? 👍

**Comment by nanspro** [comment]
any updates upon this one? @richard-ramos  @andytudhope

**Pull request merged**